.circleci/apps.yml

```diff
@@ -0,0 +1,152 @@
+
+  # Check that the CHANGELOG max line length does not exceed 80 characters
+  lint-changelog:
+    docker:
+      - image: debian:stretch
+        auth:
+          username: $DOCKER_HUB_USER
+          password: $DOCKER_HUB_PASSWORD
+    working_directory: ~/fun
+    steps:
+      - checkout
+      - run:
+          name: Check CHANGELOG max line length
+          command: |
+            # Get the longuest line width (ignoring release links)
+            test $(cat CHANGELOG.md | grep -Ev "^\[.*\]: https://github.com/openfun" | wc -L) -le 80
+
+
+  # Test the bootstrap playbook on the parametrized application
+  test-bootstrap-app:
+    # We use the machine executor, i.e. a VM, not a container
+    machine:
+      # Prevent cache-related issues
+      docker_layer_caching: false
+    working_directory: ~/fun
+
+    parameters:
+      app:
+        type: string
+
+    steps:
+      - checkout
+
+      - run:
+          name: Define Environment Variable at Runtime
+          command: |
+            make .env
+            cat .env >> $BASH_ENV
+            source $BASH_ENV
+
+      - run:
+          name: Install the kubectl client and k3d
+          command: |
+            export KUBECTL_RELEASE="v1.20.2"
+            curl -Lo "${HOME}/bin/kubectl" "https://dl.k8s.io/release/${KUBECTL_RELEASE}/bin/linux/amd64/kubectl"
+            curl -Lo /tmp/kubectl.sha256 "https://dl.k8s.io/${KUBECTL_RELEASE}/bin/linux/amd64/kubectl.sha256"
+            echo "$(</tmp/kubectl.sha256) ${HOME}/bin/kubectl" | sha256sum --check
+            chmod 755 "${HOME}/bin/kubectl"
+
+            export K3D_RELEASE="v4.2.0"
+            curl -Lo "${HOME}/bin/k3d" "https://github.com/rancher/k3d/releases/download/${K3D_RELEASE}/k3d-linux-amd64"
+            curl -sL https://github.com/rancher/k3d/releases/download/${K3D_RELEASE}/sha256sum.txt | \
+              grep _dist/k3d-linux-amd64 | \
+              sed "s|_dist/k3d-linux-amd64|${HOME}/bin/k3d|" | \
+              sha256sum --check
+            chmod 755 "${HOME}/bin/k3d"
+
+      - run:
+          name: Configure sysctl limits for Elasticsearch
+          command: |
+            # Elasticsearch requires to increase this setting's default value
+            sudo sysctl -w vm/max_map_count=262144
+
+      - run:
+          name: Run local k3d cluster & configure environment
+          command: |
+            # Bootstrap the k3d cluster with the following specific settings :
+            # - use standard HTTP and HTTPS ports
+            # - pre-provision 15 volumes instead of 100
+            MINIMUM_AVAILABLE_RWX_VOLUME=15 \
+            K3D_BIND_HOST_PORT_HTTP=80 \
+            K3D_BIND_HOST_PORT_HTTPS=443 \
+              make cluster
+
+      - run:
+          name: Test the "<< parameters.app >>" application bootstrapping
+          command: |
+            # Bootstrap app
+            bin/arnold -v ${PWD}/apps:/app/apps -c eugene -e ci -a << parameters.app >> bootstrap
```

Comment:
After :thinking:, I share with you my thoughts on this subject.
I'm wondering if this generic test will be suitable for testing all the applications of this repository.

Is testing the `bootstrap` phase of an application sufficient to ensure that it works properly?
We will probably want to run specific tests for certain applications (e.g. checking the reachability of a service and testing the response content).

And with this way of declaring jobs, I don't see how we can create an application-specific test.
WDYT?

Reply:
I asked myself the exact same questions about having a generic way of testing our trays. My conclusions about these follows:

1. most tests wrote for Arnold-bundled trays are in fact testing Arnold internals rather than created objects (I'm thinking of systematic switch and rollback tests),
2. testing an application bootstrapping seems "good-enough" to cover a large part of issues we may encounter with a tray,
3. if needed (but I don't think it is), refactor this CI to switch to a base YAML template for applications that can be further customized per-app is totally feasible (but for now out-of-scope for this PR),
4. the perfect way to test trays would be to use an end-to-end test engine such as Cypress for web applications, but IMO it's a bit overkill...

An alternative test suite that can be generic would be to test:

1. Application pod statuses
2. Created services reachability

Reply:
Thanks for sharing your conclusions! :+1: 
I'm reassured by your point `3`, because I'm convinced it will be necessary. Not today, but in a near future :slightly_smiling_face: 

I also think e2e testing is overkill here.

Reply:
> most tests wrote for Arnold-bundled trays are in fact testing Arnold internals rather than created objects (I'm thinking of systematic switch and rollback tests),

I think we are really testing if the application is available after a deploy. For sure we also test arnold internals but the mainpurpose is to be sure the application is deployed and available, if not there is a mistake somewhere in our templates.

So I agree with @madmatah , we will need to implement specific test for a specific app. Is it possible to create conditional job based on `parameters.app` value ?

Reply:
> the mainpurpose is to be sure the application is deployed and available

To me this should be ensured by pods probes, and this is Arnold's responsibility to check readiness status of your application. So, to me we must improve Arnold's wait loop to ensure that is checks pods readiness status (and not only that it has already started), and systematically test deployed services response.